presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -342,4 +352,30 @@ public static ReaderPageSourceWithProjections createPageSource(
         }
         return null;
     }
+
+    private static Map<List<String>, RichColumnDescriptor> getDescriptorsUsingIndex(List<HiveColumnHandle> hiveColumnHandles, MessageType fileSchema)
+    {
+        Map<List<String>, RichColumnDescriptor> descriptorsByName = new HashMap<>();
```

Comment:
- descriptorsByName -> descriptorsByColumnName 
- move this line just before the `for` loop where it's being populated
- and use ImmutableList.builder